### ISSUE TITLE
Support vim's flavor of regex

### ISSUE THREAD

**Issue by J-Fields** [VSCodeVim/Vim#3996]
We currently use javascript's regex library, which is unfamiliar to some coming from vim (see #2043, #3073, #3075, etc.). This ticket is to consolidate those and center discussion around the possibility of supporting vim's regex flavor, though at this point there are no plans to do so.

**Comment by Banger** [comment]
Very interested in this.  The whole point of using the VIM extension is so I don't have to learn yet another editor and its quickiness.
You should look into VIM implementations in javascript.  There are a lot (not many good ones though).

**Comment by FrankCoder** [comment]
If this ever gets integrated, I believe it would be a good to have the option.  I've been heavily using a variety of regex engines for more years than I care to say and I always thought Vim's regex syntax was clunky.

**Comment by sql-koala** [comment]
I think there is another option with less effort than a full regex engine. I am stealing ;-) this idea from vsvim. What the guys there did, is a "tokenwise" translation. Meaning:
```
\<  -> \b
\l  -> [a-z]
```
And so on... A lot of vim regex could be translated like so. Limits are where functionality is principally missing in js regex.

**Comment by J-Fields** [comment]
Yeah that's a good point, @sql-koala, I've pondered it before. There are definitely limitations, but it'd be a good first solution I think.

**Comment by JohnnyUrosevic** [comment]
Hey guys, I imported a .vimrc which has `nnoremap / /\v` and learned about the concept of 'very magic' in vim, is there any plans to add this in addition to magic / no magic?

**Comment by J-Fields** [comment]
> Is there any plans to add this in addition to magic / no magic?

Yes, support for `magic` is planned, though it may be a while before it's implemented

**Comment by JohnnyUrosevic** [comment]
> 
> 
> > Is there any plans to add this in addition to magic / no magic?
> 
> Yes, support for `magic` is planned, though it may be a while before it's implemented

I can try to work on this if you want. I don't know if there's more pressing issues you would rather have contributors for.

**Comment by macintacos** [comment]
Just noting here that with the new (fantastic!) inccommand addition, this issue makes it all the more puzzling when it seems like the RegEx matches, the UI updates, but it doesn't actually do anything because the RegEx doesn't match when doing the replace (from what I can tell). See this comment: https://github.com/VSCodeVim/Vim/issues/7499#issuecomment-1039755795

**Comment by rdbisme** [comment]
`s:(\S+):"\1":g` shows the replacement correctly in the UI, but then it provides `E486: Pattern not found` when hitting enter.

**Comment by ahmafi** [comment]
`%s/stroke=".*?"/stroke="currentColor"` also highlights correctly in the editor, but then it shows `E486: Pattern not found` when I try to replace.
